kinappserver/tests/app_discovery.py

```diff
@@ -0,0 +1,207 @@
+            'category_id': '0',
+            'title': 'cat-title'
+        }
+        # add category to db
+        resp = self.app.post('/app_discovery/add_discovery_app_category',
+                             data=json.dumps({
+                                 'discovery_app_category': cat}),
+                             headers={},
+                             content_type='application/json')
+        self.assertEqual(resp.status_code, 200)
+
+        # test if user sees the category
+        resp = self.app.get('/app_discovery', headers={USER_ID_HEADER: str(android_user)})
+        self.assertEqual(resp.status_code, 200)
+
+        print(json.loads(resp.data))
+        data = json.loads(resp.data)
+
+        # test that category data is correct
+        self.assertEqual(data[0]['category_id'], 0)
+        self.assertEqual(data[0]['title'], 'cat-title')
+        self.assertEqual(data[0]['apps'], [])
+
+        # create an android app
+        app = {
+            'skip_image_test': 'true',
+            'app_identifier': '0',
+            'title': 'app title',
+            'subtitle': 'subtitle',
+            'app_category_id': '0',
+            'os_type': 'android',
+            'meta_data': {
+                'app_url': 'url',
+                'description': 'long desc',
+                'kin_usage': 'how to use description',
+                'icon_url': 'url',
+                'card_image_url': 'url',
+                'image_url': 'url',
+            }
+        }
+        # add the app
+        resp = self.app.post('/app_discovery/add_discovery_app',
+                             data=json.dumps({
+                                 'discovery_app': app,
+                                 'set_active': 'True'}),
+                             content_type='application/json')
+        self.assertEqual(resp.status_code, 200)
+
+        # verify android user sees the app
+        resp = self.app.get('/app_discovery', headers={USER_ID_HEADER: str(android_user)})
+        self.assertEqual(resp.status_code, 200)
+
+        print(json.loads(resp.data))
+        data = json.loads(resp.data)
+
+        self.assertEqual(data[0]['category_id'], 0)
+        self.assertEqual(data[0]['title'], 'cat-title')
+        self.assertEqual(len(data[0]['apps']), 1)
+
+        # verify iOS user does not see the app
+        resp = self.app.get('/app_discovery', headers={USER_ID_HEADER: str(ios_user)})
+        self.assertEqual(resp.status_code, 200)
+
+        print(json.loads(resp.data))
+        data = json.loads(resp.data)
+
+        self.assertEqual(data[0]['category_id'], 0)
+        self.assertEqual(data[0]['title'], 'cat-title')
+        self.assertEqual(len(data[0]['apps']), 0)
+
+        # create an androdi and ios app
+        app = {
+            'skip_image_test': 'true',
+            'app_identifier': '1',
+            'title': 'app title',
+            'subtitle': 'subtitle',
+            'app_category_id': '0',
+            'os_type': 'android, iOS',
```

Comment:
how would that work ? we need an additional app_url in the metadata in such cases